src/katgpucbf/xbgpu/correlation.py

```diff
@@ -79,30 +81,33 @@ def __init__(
         n_ants: int,
         n_channels: int,
         n_spectra_per_heap: int,
-        sample_bitwidth: int,
+        input_sample_bits: int,
     ) -> None:
+        if input_sample_bits != 8:
+            raise ValueError(f"input_sample_bits of 8 is only supported, given {input_sample_bits}.")
+
```

Comment:
I know I'd suggested moving this in here, but it's actually redundant with the exceptions at lines 104-111. Delete one of them (maybe this one, since the existing one is more explicit about whether the value is unimplemented or just wrong.

Reply:
Too right.